modules/oe_theme_content_publication/src/Plugin/ExtraField/Display/PublicationCollection.php

```diff
@@ -0,0 +1,154 @@
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Url;
+use Drupal\extra_field\Plugin\ExtraFieldDisplayFormattedBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Displays publication collections.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_publication_collection",
+ *   label = @Translation("Part of collection"),
+ *   bundles = {
+ *     "node.oe_publication",
+ *   },
+ *   visible = true
+ * )
+ */
+class PublicationCollection extends ExtraFieldDisplayFormattedBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The field label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * PublicationCollection constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->label;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    $build = [];
+    /** @var \Drupal\node\Entity\Node[] $collections */
+    $collections = $this->getReferencingCollections($entity);
+
+    if (empty($collections)) {
+      return $build;
+    }
+
+    $build = [
+      '#theme' => 'oe_theme_content_publication_collections',
+    ];
+
+    // Set the label of the field.
+    if (count($collections) > 1) {
+      $this->label = $this->t('Part of collections');
+    }
+    else {
+      $this->label = $this->t('Part of collection');
+    }
+
+    $cache = CacheableMetadata::createFromRenderArray($build);
+
+    $items = [];
+    foreach ($collections as $collection) {
+      $cache->addCacheableDependency($collection);
+      $items[] = [
```

Comment:
Yes, but we are in extra field context and not drupal field that template was made for. We can keep the custom template instead of hack the render array to be able to use the field--multiple-values--pipe-separated.html.twig.
On the other hand, we need an url object there, and that line I took from core entity reference revisions label formatter.

Reply:
`toUrl()` also returns Url object and handle edge cases in node.
As I see https://git.drupalcode.org/project/drupal/-/blob/9.3.x/core/lib/Drupal/Core/Field/Plugin/Field/FieldFormatter/EntityReferenceLabelFormatter.php#L68 `$entity->toUrl()` is used and wrapped by try/catch.